client/packages/core/src/datalog.js

```diff
@@ -17,39 +17,43 @@ function matchExact(patternPart, triplePart, context) {
   return patternPart === triplePart ? context : null;
 }
 
-function matchWithArgMap(patternPart, triplePart, context) {
-  const { in: inList, $in: $inList } = patternPart;
-  if (
-    (inList && inList.includes(triplePart)) ||
-    ($inList && $inList.includes(triplePart))
-  ) {
-    return context;
-  }
-
-  if (
-    patternPart.hasOwnProperty("$not") ||
-    patternPart.hasOwnProperty("$isNull")
-  ) {
-    // If we use `$not` or `$isNull`, we've already done the filtering in
-    // `getTriples`
-    return context;
-  }
-  return null;
-}
-
 function matcherForPatternPart(patternPart) {
   switch (typeof patternPart) {
     case "string":
       return patternPart.startsWith("?") ? matchVariable : matchExact;
-    case "object":
```

Comment:
We already do all the filtering in the store, so this saves us an extra comparison on the data.